crates/astria-sequencer-relayer/src/relayer/write/mod.rs

```diff
@@ -269,15 +297,56 @@ impl BlobSubmitter {
     }
 }
 
+/// Tries to confirm the last attempted submission of the previous session.
+///
+/// This should only be called where submission state on startup is `Prepared`, meaning we don't yet
+/// know whether that final submission attempt succeeded or not.
+///
+/// Internally, this polls `GetTx` for up to one minute.  The returned `SubmissionState` is
+/// guaranteed to be in `Finished` state, either holding the heights of the previously prepared
+/// submission if confirmed by Celestia, or holding the heights of the last known confirmed
+/// submission in the case of timing out.
+#[instrument(skip_all)]
+async fn try_confirm_submission_from_last_session(
+    mut client: CelestiaClient,
+    prepared_submission: PreparedSubmission,
+    state: Arc<super::State>,
+    metrics: &'static Metrics,
+) -> eyre::Result<StartedSubmission> {
+    let blob_tx_hash = prepared_submission.blob_tx_hash();
+    info!(%blob_tx_hash, "confirming submission of last `BlobTx` from previous session");
+
+    let timeout = prepared_submission.confirmation_timeout();
+    let new_state =
+        if let Some(celestia_height) = client.try_confirm_submission(blob_tx_hash, timeout).await {
+            info!(%celestia_height, "confirmed previous session submitted blobs to Celestia");
+            prepared_submission.into_started(celestia_height).await?
```

Comment:
please add context

Reply:
Done in [9ade24c](https://github.com/astriaorg/astria/pull/1234/commits/9ade24cf8a1b470ef1ba8524afad2fcbab2dd4ab).